tests/message-extensions/src/index.ts

```diff
@@ -0,0 +1,129 @@
+import { cardAttachment } from '@microsoft/teams.api';
+import { App } from '@microsoft/teams.apps';
+import { Card } from '@microsoft/teams.cards';
+import { ConsoleLogger } from '@microsoft/teams.common/logging';
+import { DevtoolsPlugin } from '@microsoft/teams.dev';
+import {
+  createCard,
+  createConversationMembersCard,
+  createDummyCards,
+  createLinkUnfurlCard,
+  createMessageDetailsCard,
+} from './card';
+
+const app = new App({
+  logger: new ConsoleLogger('@tests/message-extensions', { level: 'debug' }),
+  plugins: [new DevtoolsPlugin()],
+});
+
+app.on('message', async ({ send, activity }) => {
+  await send({ type: 'typing' });
+  await send(`you said "${activity.text}"`);
+});
+
+// :snippet-start: message-ext-query-link
+app.on('message.ext.query-link', async ({ activity }) => {
+  const { url } = activity.value;
+
+  if (!url) {
+    return {
+      status: 400,
+      body: {},
+    };
+  }
+
+  const { card, thumbnail } = createLinkUnfurlCard(url);
+  const attachment = {
+    ...cardAttachment('adaptive', card),
+    preview: cardAttachment('thumbnail', thumbnail),
+  };
+
+  return {
+    composeExtension: {
+      type: 'result',
+      attachmentLayout: 'list',
+      attachments: [attachment],
+    },
+  };
+});
+// :snippet-end: message-ext-query-link
+// :snippet-start: message-ext-submit
+app.on('message.ext.submit', async ({ send, activity }) => {
+  const { commandId } = activity.value;
+  let card: Card;
+
+  if (commandId === 'createCard') {
+    // activity.value.commandContext == "compose"
+    card = createCard(activity.value.data);
+  } else if (commandId === 'getMessageDetails' && activity.value.messagePayload) {
+    // activity.value.commandContext == "message"
+    card = createMessageDetailsCard(activity.value.messagePayload);
+  } else {
+    throw new Error(`Unknown commandId: ${commandId}`);
+  }
+
+  await send(card);
+
+  return {
+    status: 200,
+    body: {},
+  };
+});
+// :snippet-end: message-ext-submit
+
+// :snippet-start: message-ext-open
+app.on('message.ext.open', async ({ activity, api }) => {
+  const conversationId = activity.conversation.id;
+  const members = await api.conversations.members(conversationId).get();
+  const card = createConversationMembersCard(members);
+
+  return {
+    status: 200,
+    body: {
+      task: {
```

Comment:
no need to return status: 200/body for successful responses. simply return { task {... }}

Reply:
fixed